### PR TITLE
feat: added functionality for jest custom ruleset

### PR DESCRIPTION
**Pull request by jaig-0911**

### CODE REVIEW

File: cSpell.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
         "wcag",
         "wdio",
         "webdriverio",
-        "webm"
+        "webm",
+        "Vidyard"
```
Comment (navateja-alagam) [inline cSpell.json:18]: Why did we add this?

File: packages/common/src/helpers.ts [hunk 2]
```diff
@@ -32,3 +33,20 @@ export function useFilesToBeExempted(): string[] {
     }
     return [];
 }
+
+export function useCustomRules(): string[] {
+    const filePath = process.env.SA11Y_CUSTOM_RULES ?? '';
+    if (filePath !== '') {
+        try {
+            // Read the file synchronously
+            const data = fs.readFileSync(filePath, 'utf-8');
```
Comment (navateja-alagam) [inline packages/common/src/helpers.ts:42]: please use async file read

File: packages/common/src/helpers.ts [hunk 3]
```diff
@@ -32,3 +33,20 @@ export function useFilesToBeExempted(): string[] {
...
+            const jsonData = JSON.parse(data) as { rules: string[] };
+            // Access the rules array
+            const rules = jsonData.rules;
```
Comment (navateja-alagam) [inline packages/common/src/helpers.ts:45]: we can minimize this 2 lines of code to one
`const { rules } = JSON.parse(data) as { rules: string[] };`

File: packages/jest/src/automatic.ts [hunk 4]
```diff
@@ -91,7 +92,12 @@ export async function automaticCheck(opts: AutoCheckOpts = defaultAutoCheckOpts)
             //      : ${testPath}`
             // );
             // W-10004832 - Exclude descendancy based rules from automatic checks
-            violations.push(...(await getViolationsJSDOM(currNode, adaptA11yConfig(defaultRuleset))));
+            if (customRules.length == 0)
```
Comment (navateja-alagam) [inline packages/jest/src/automatic.ts:95]:
```suggestion
            if (customRules.length === 0)
```

File: cSpell.json (hunk 1 above)
Comment (jaig-0911) [reply cSpell.json:18]: When I was trying to commit, it gave me a spell check error on "Vidyard" word. So I added it here.

File: packages/common/src/helpers.ts (hunk 2 above)
Comment (jaig-0911) [reply packages/common/src/helpers.ts:42]: I tried that but It gives a warning "(!) Missing shims for Node.js built-ins
Creating a browser bundle that depends on "fs/promises".

**Pull request merged**